src/wayland-egldisplay.c

```diff
@@ -500,6 +502,95 @@ static const struct wl_drm_listener drmListener = {
 };
 
 
+static void
+dmabuf_feedback_check_main_device(void *data,
+                            struct zwp_linux_dmabuf_feedback_v1 *dmabuf_feedback,
+                            struct wl_array *dev)
+{
+    WlServerProtocols *protocols = (WlServerProtocols *)data;
+    (void) dmabuf_feedback;
+
+    assert(dev->size == sizeof(dev_t));
+    memcpy(&protocols->devId, dev->data, sizeof(dev_t));
+}
+
+static void
+dmabuf_feedback_check_tranche_target_device(void *data,
+                                      struct zwp_linux_dmabuf_feedback_v1 *dmabuf_feedback,
+                                      struct wl_array *dev)
+{
+    (void) data;
+    (void) dmabuf_feedback;
+    (void) dev;
+}
+
+static void
+dmabuf_feedback_check_tranche_flags(void *data,
+                              struct zwp_linux_dmabuf_feedback_v1 *dmabuf_feedback,
+                              uint32_t flags)
+{
+    (void) data;
+    (void) dmabuf_feedback;
+    (void) flags;
+}
+
+static void
+dmabuf_feedback_check_tranche_formats(void *data,
+                                struct zwp_linux_dmabuf_feedback_v1 *dmabuf_feedback,
+                                struct wl_array *indices)
+{
+    (void) data;
+    (void) dmabuf_feedback;
+    (void) indices;
+}
+
+static void
+dmabuf_feedback_check_tranche_done(void *data,
+                             struct zwp_linux_dmabuf_feedback_v1 *dmabuf_feedback)
+{
+    (void) data;
+    (void) dmabuf_feedback;
+}
+
+static void
+dmabuf_feedback_check_done(void *data, struct zwp_linux_dmabuf_feedback_v1 *dmabuf_feedback)
+{
+    WlServerProtocols *protocols = (WlServerProtocols *)data;
+    (void) dmabuf_feedback;
+
+    assert(protocols->devId != 0);
+    drmDevice *drm_device;
+    if (drmGetDeviceFromDevId(protocols->devId, 0, &drm_device) == 0) {
```

Comment:
This is a relatively new function in libdrm. I don't see a particular version of libdrm required in the  current meson.build. Perhaps we should add a requirement for version 2.4.109, the first version to include this function?

Reply:
Are there any systems that are both new enough to support Wayland and old enough to not have `drmGetDeviceFromDevId`?

If so, then we'd probably need to look up `drmGetDeviceFromDevId` at runtime using dlsym.

Reply:
2.4.109 was released May of 2021, so maybe? But if we're going to do that would it be easier to just return `"/dev/dri/renderD" + minor(protocols->devId)`? It looks like that's pretty much how libdrm populates `drmDevice::nodes`.

Reply:
Even though I think that's reliable in practice, I don't think we're supposed to rely on file layout assumptions. It probably is safe to assume systems without DRM 2.4.109 are running compositors that still support wl_drm though, so a dynamic lookup and skipping this path if it isn't there seems fine. Normally I'd say just compile it out if the DRM version is too old, but since we'll have to ship this as a binary in our driver package as well, I agree a runtime check is needed.

Reply:
I updated the code to use dlsym now